blog/sage-integrating-rainbow-kit-celo-into-your-dapps/sage-integrating-rainbow-kit-celo-into-your-dapps.md

```diff
@@ -0,0 +1,752 @@
+      <h1 className="mb-4 font-semibold text-2xl">
+        Simple Will Smart Contract
+      </h1>
+      <ConnectButton />
+      <div className="mt-8">
+        <div className="bg-white border border-green-500 border-solid p-4 rounded-lg text-center">
+          {displayValue ? displayValue : ""}
+        </div>
+        <div className="mt-4 grid grid-cols-3 gap-8">
+          <button className="bg-red-400 py-2 px-4 text-white rounded-lg">
+            Change Owner
+          </button>
+          <button className="bg-red-400 py-2 px-4 text-white rounded-lg">
+            Set Heir
+          </button>
+          <button className="bg-red-400 py-2 px-4 text-white rounded-lg">
+            Change Heir
+          </button>
+          <button className="bg-red-400 py-2 px-4 text-white rounded-lg">
+            Change Deceased Status
+          </button>
+          <button
+            onClick={getOwner}
+            className="bg-green-400 py-2 px-4 text-white rounded-lg"
+          >
+            Owner
+          </button>
+          <button onClick={getHeir} className="bg-green-400 py-2 px-4 text-white rounded-lg">
+            Heir
+          </button>
+          <button onClick={getStatus} className="bg-green-400 py-2 px-4 text-white rounded-lg">
+            Status
+          </button>
+          <button className="bg-red-600 py-2 px-4 text-white rounded-lg">
+            Transfer Inheritance
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export default App;
+```
+
+In the code above we have a UI that has the `ConnectButton` provided by `@rainbow-me/rainbowkit` rendered. This custom button allows us to connect to default crypto wallets provided by Celo and other crypto wallets that support Celo. 
+
+The `getOwner()`, `getHeir(`) and `getStatus()`  are functions that are called when a user clicks a particular button. These functions connect with the smart contract we created and deployed to call the various methods we created in the smart contract.
+
+Start up the React app by running this command in your terminal to view the Dapp
+
+```bash
+yarn start
+```
+
+Make sure the terminal points to the dapp folder before running the command.
+
+This is how the Dapp should look when the application opens
+
+<img width="960" alt="will-Dapp" src="https://user-images.githubusercontent.com/104994589/223736033-b77b2163-60bf-49b4-9aba-bb49ad2a3f2c.png">
+
+## Conclusion
+
+So far, we have been able to create a smart contract Dapp using React. We created the UI enabling a user to connect their wallet to the Dapp using Rainbowkit-celo. We also connected with the smart contract we deployed from the Dapp and called some of the methods we created in the smart contract.
+
+## Next Steps/References
+
+If you wish to read more on rainbowkit and Rainbowkit-celo, check out this docs. [the official github page of Rainbowkit-celo](https://github.com/celo-org/rainbowkit-celo), [rainbowkit docs](https://www.rainbowkit.com/docs/introduction)
```

Comment:
how bout this 
can you change it to this instead (just the structuring)
If you wish to read more on rainbowkit and Rainbowkit-celo, check out this docs. 
* [the official github page of Rainbowkit-celo](https://github.com/celo-org/rainbowkit-celo) 
* [rainbowkit docs](https://www.rainbowkit.com/docs/introduction)